renderdoc/driver/metal/metal_init_state.cpp

```diff
@@ -39,16 +102,44 @@ bool WrappedMTLDevice::Prepare_InitialState(WrappedMTLObject *res)
 
 uint64_t WrappedMTLDevice::GetSize_InitialState(ResourceId id, const MetalInitialContents &initial)
 {
-  METAL_NOT_IMPLEMENTED();
-  return 128;
+  uint64_t ret = 12;
+
+  if(initial.type == eResBuffer)
+  {
+    ret += uint64_t(8 + initial.resourceContents.size() + WriteSerialiser::GetChunkAlignment());
+    return ret;
+  }
+
+  RDCERR("Unhandled resource type %s", ToStr(initial.type).c_str());
+  return 0;
 }
 
 template <typename SerialiserType>
 bool WrappedMTLDevice::Serialise_InitialState(SerialiserType &ser, ResourceId id,
                                               MetalResourceRecord *record,
                                               const MetalInitialContents *initial)
 {
-  METAL_NOT_IMPLEMENTED();
+  SERIALISE_ELEMENT_LOCAL(type, initial->type);
+  SERIALISE_ELEMENT(id).TypedAs(NameOfType(type)).Important();
+  if(type == eResBuffer)
+  {
+    SERIALISE_CHECK_READ_ERRORS();
+
+    bytebuf localContents;
+    bytebuf &contents(localContents);
```

Comment:
This is fine but reads a little oddly to me at first glance. For initialising references/pointers like this I'd prefer `bytebuf &contents = localContents;`.